src/ORM/ResultSet.php

```diff
@@ -410,7 +411,14 @@ protected function _calculateColumnMap($query)
             }
 
             $parts = explode('__', $key, 2);
-            $map[$parts[0]][$key] = $parts[1];
+            // In PostgreSQL, identifiers — table names, column names, constraint names, etc. — are limited to a
+            // maximum length of 63 bytes. Identifiers longer than 63 characters can be used, but they will be truncated
+            // to the allowed length of 63.
+            if ($this->_driver instanceof Postgres) {
+                $map[$parts[0]][substr($key, 0, 63)] = $parts[1];
```

Comment:
@ADmad / @ndm2 what do you think?

Primarly stumpled upon this when converting a cakephp application from mariadb to postgresql.

How are you handling table and column names in case they get to long vs. cakephp naming conventions?